src/lib/oauth.ts

```diff
@@ -67,32 +66,37 @@ export async function isAuthenticated(): Promise< boolean > {
 	return !! token;
 }
 
-export async function handleAuthCallback( hash: string ): Promise< Error | StoredToken > {
+async function handleAuthCallback( hash: string ): Promise< StoredToken > {
 	const params = new URLSearchParams( hash.substring( 1 ) );
 	const error = params.get( 'error' );
+
 	if ( error ) {
 		// Close the browser if code found or error
-		return new Error( error );
+		throw new Error( error );
 	}
+
 	const accessToken = params.get( 'access_token' ) ?? '';
 	const expiresIn = parseInt( params.get( 'expires_in' ) ?? '0' );
+
 	if ( isNaN( expiresIn ) || expiresIn === 0 || ! accessToken ) {
-		return new Error( 'Error while getting token' );
+		throw new Error( 'Error while getting token' );
 	}
-	let response: { ID?: number; email?: string; display_name?: string } = {};
+
 	try {
-		response = await new wpcom( accessToken ).req.get( '/me?fields=ID,email,display_name' );
+		const response = await new wpcom( accessToken ).req.get( '/me?fields=ID,email,display_name' );
+
```

Comment:
@fredrikekelund I guess here would be the new point to specifically log it to sentry if `ID` is ever returned empty from api endpoint unless whatever zod will throw would capture that info?

Reply:
We already have a `Sentry.captureException` call in the catch clause, and the nice thing about zod's `parse` function is that it includes a clear explanation of the validation error in the error **message**. Here's an example:
```
[
  {
    "code": "invalid_type",
    "expected": "string",
    "received": "null",
    "path": [
      "accessToken"
    ],
    "message": "Expected string, received null"
  }
]
```